velox/type/DecimalUtil.h

```diff
@@ -250,8 +250,24 @@ class DecimalUtil {
     return (unsignedSum >> 127);
   }
 
+  /// Adds two signed 128-bit numbers (int128_t), calculates the sum, and
+  /// returns the overflow. It can be used to track the number of overflow when
+  /// adding a batch of input numbers. It takes lhs and rhs as input, and stores
+  /// their sum in result. overflow == 1 indicates upward overflow. overflow ==
+  /// -1 indicates downward overflow. overflow == 0 indicates no overflow.
+  /// Adding negative and non-negative numbers never overflows, so we can
+  /// directly add them. Adding two negative or two positive numbers may
+  /// overflow. To add numbers that may overflow, first convert both numbers to
+  /// unsigned 128-bit number (uint128_t), and perform the addition. The highest
+  /// bits in the result indicates overflow. Adjust the signs of sum and
+  /// overflow based on the signs of the inputs. After following the above logic
+  /// to sum a batch of input numbers, if the final accumulated overflow and sum
+  /// meet one of the following conditions,
+  /// 1. overflow == 1 && result < 0
+  /// 2. overflow == -1 && result > 0
+  /// we should use the adjustSumForOverflow method below to adjust the result.
   inline static int64_t
-  addWithOverflow(int128_t& result, const int128_t& lhs, const int128_t& rhs) {
+  addWithOverflow(int128_t& result, int128_t lhs, int128_t rhs) {
```

Comment:
Let's change the signature to

```
int128_t addWithOverflow(int128_t a, int128_t b, int64_t& overflow)
```

Let's have this function return the same and set 'overflow' to one of (0, 1, -1).

Reply:
Sure. I will ping you when I finish.

Reply:
@mbasmanova I realized during the modification that the reason is that during the update process of the sum aggregation, `accumulator->sum` can be directly overwritten, so it is used as a reference. `accumulator->overflow` is not overwritten but accumulated. You can take a look at this code,
```cpp
overflow += DecimalUtil::addWithOverflow(
    this->sum, HugeInt::build(upperSum, lowerSum), this->sum);
```
If addWithOverflow return sum while overflow is a reference, the code would become more complex,
```cpp
int64_t sumOverflow;
this->sum = DecimalUtil::addWithOverflow(
    this->sum, HugeInt::build(upperSum, lowerSum), sumOverflow);
this->overflow += sumOverflow;
```
Should we maintain the original function signature?

Reply:
@liujiayi771 I see. Thank you for explaining. It makes sense now. We can keep it this way or maybe streamline a bit:

```
int64_t addWithOverflow(int128_t& to, int128_t value)
```

We can take just 2 parameters: current sum (to) and value to add to it (value) and return overflow. Then usage will be a bit like sum += value:

```
overflow += DecimalUtil::addWithOverflow(sum, HugeInt::build(upperSum, lowerSum));
```

Reply:
@mbasmanova DecimalUtil::computeAverage does not fit to this pattern.
```cpp
DecimalUtil::addWithOverflow(avg, sumA, sumB);
```